src/index.ts

```diff
@@ -1,8 +1,27 @@
 import realm from 'realm'
+import sift from 'sift'
 
 const aggregate = (aggregate: any[], realm: realm, model: string ) => {
+	return aggregate.reduce((prevVal, curVal, index, array) => {
+		return applyAggregation(array[index], prevVal);
```

Comment:
It's a bit of a hack.  `curVal` will initially be the first aggregation object, but after that, it will contained the modified collection object.  
But having said this, I see that this currently accidentally works.  On the second run `prevVal` will contain the aggregation object and that will cause some bugs :) 

I will refactor this so that it's more clear and correct.

Reply:
Ah yeah, I agree. 
I initially got tricked by the `prevVal` and forgot about `accumulator` passed as the first argument to the callback.